src/battle_util.c

```diff
@@ -6793,6 +6793,10 @@ static u8 ItemHealHp(u32 battler, u32 itemId, bool32 end2, bool32 percentHeal)
             BattleScriptPushCursor();
             gBattlescriptCurrInstr = BattleScript_ItemHealHP_RemoveItemRet;
         }
+        if (gBattleResources->flags->flags[battler] & RESOURCE_FLAG_EMERGENCY_EXIT
+         && GetNonDynamaxMaxHP(battler) > gBattleMons[battler].maxHP / 2)
```

Comment:
why are we comparing two maxHp values here? Shouldnt this be `GetNonDynamaxHP(battler) > gBattleMons[battler].maxHP/2`?

Reply:
i messed up. Fixed in  #4041 follow up.